content/200-concepts/100-components/02-prisma-client/260-client-extensions/130-result.mdx

```diff
@@ -1,16 +1,159 @@
 ---
-title: 'Add custom results objects to Prisma Client'
-metaTitle: 'Prisma Client extensions: result (Preview)'
+title: 'Add custom fields and methods to query results'
+metaTitle: 'Prisma Client extensions: result component (Preview)'
 metaDescription: 'Extend the functionality of Prisma Client'
 tocDepth: 3
 ---
 
 <TopBlock>
 
-<Admonition type="warning">
+From version 4.7.0, you can use the `result` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) component to add custom fields and methods to query results.
 
-We plan to add `result` extensions to a future version of Prisma. This page is a placeholder.
+</TopBlock>
 
-</Admonition>
+## Enable the preview feature
 
-</TopBlock>
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Add custom fields or methods to query results
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an _extended client_. An extended client is a variant of the standard Prisma Client that is wrapped by one or more extensions.
+
+Use the `result` extension component to add custom fields and methods to query results.
+
+To add a custom [field](#add-a-custom-field-to-the-results-object) or [method](#add-a-custom-method-to-the-results-object) to query results, use the following structure. In this example, we add a custom field to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  result?: {
+    user: {     // in this case, we extend the `user` model
+      needs: { ... }
+      fields: { ... }
+    },
+  },
+});
```

Comment:
Do we need a `compute` statement here? Is `compute` always. component?

@millsp @SevInf 

For Andrew to do: If so, we need to mention it
- above
- in the parameters below

Reply:
`needs` and `fields` are old syntax from proposal, the way it is actually implemented is

```ts
result: {
   user: {
      myComputedField: {
          needs: {},
          compute() { ... }
      }
   }
}
```

We should change all examples where `fields` property occurs to this syntax.

Reply:
Thanks @SevInf